integration-tests/src/test/scala/chiselTest/util/SparseVecSpec.scala

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.util
+
+import chisel3._
+import chisel3.stage.WarningConfigurationAnnotation
+import chisel3.testers.BasicTester
+import chisel3.util.{log2Up, Counter, SparseVec}
+import chiselTests.{ChiselFlatSpec, Utils}
+import _root_.circt.stage.ChiselStage
+
+// This test is checking that a SparseVec behaves _exactly_ like a dynamic
+// index.  By varying the mapping and size, different corner cases of behavior
+// are checked.
+class SparseVecTest(size: Int, tpe: UInt, mapping: Seq[(Int, UInt)]) extends BasicTester {
+  private val paddedSize = BigInt(log2Up(size)).pow(2).toInt
+
+  // This is the reference vector that will be dynamically indexed into.
+  // Initialize all elements to DontCare.  Then set specific ones using the
+  // provided mapping.
+  private val denseVec = {
+    val w = Wire(Vec(size, tpe))
+    w.foreach(_ := DontCare)
+    mapping.foreach {
+      case (index, data) =>
+        w(index) := data
+    }
+    w
+  }
+
+  // Create a wire SparseVec and initialize it to the values in the mapping.
+  private val sparseVec = Wire(new SparseVec(size, tpe, mapping.map(_._1)))
+  sparseVec.elements.values.zip(mapping.map(_._2)).foreach { case (a, b) => a :<>= b.asInstanceOf[Data] }
+
+  // Access the dense vector and the sparse vector, using all of the access
+  // types, and make sure that the results are exactly the same.
+  private val (index, wrap) = Counter(0 until paddedSize)
+  private val failed = RegInit(Bool(), false.B)
+  private val reference = denseVec(index)
+  private val sparseVecResults = Seq(SparseVec.Binary, SparseVec.OneHot, SparseVec.IfElse).map(sparseVec(index, _))
+  when(sparseVecResults.map(_ =/= reference).reduce(_ || _)) {
+    failed := true.B
+    assert(false.B)
+  }
+
+  when(wrap) {
+    stop()
+  }
+
+}
+
+class SparseVecSpec extends ChiselFlatSpec with Utils {
+  "SparseVec" should "work for a complete user-specified mapping" in {
+    assertTesterPasses(
+      new SparseVecTest(
+        4,
+        UInt(3.W),
+        Seq(
+          0 -> 1.U,
+          1 -> 2.U,
+          2 -> 3.U,
+          3 -> 4.U
+        )
+      ),
+      // TODO: This warning suppression doesn't work for some reason.
+      annotations = Seq(WarningConfigurationAnnotation("id=4:s"))
```

Comment:
You should be able to fix the warning by making the indices the correct widths

Reply:
Fixed in e4e076076.